.github/workflows/main.yml

```diff
@@ -46,10 +48,24 @@ jobs:
           key: go-artifacts-${{ runner.os }}-check-all-${{ steps.go.outputs.go-version }}-${{ hashFiles('go.sum', 'Makefile', 'app/**/Makefile') }}
           restore-keys: go-artifacts-${{ runner.os }}-check-all-
 
-      - name: Run check-all
-        run: |
-          make check-all
-          git diff --exit-code
+      - name: Run fmt
+        run: make fmt
+
+      - name: Run vet
+        run: make vet
+      
+        # Use action instead of `make golangci-lint` to speed up the process
+        # as it caches data between builds.
+      - name: Run golangci-lint
+        uses: golangci/golangci-lint-action@v6
+        with:
+          version: v1.59
```

Comment:
this change introduces discrepancy for local and CI commands.

golangci-lint version requires to be in sync with Makefile and github ci. As far as I remember we already had such problem and decided to use only make file commands https://github.com/VictoriaMetrics/VictoriaMetrics/commit/c560a338e8c41b1854b1e0bdb4ce2697ddf68956

Reply:
I've fixed the mismatch and added comments in each part to remind keeping it in sync.
I guess it's something we can live with as a tradeoff for more performant lint process? Let me know if I'm wrong.